### PR TITLE
Autocomplete and UI

### PR DESCRIPTION
**Pull request by RamyaPrabakar**
**Quick summary of change**
Implemented autocomplete and made some UI changes. Also, added a confetti animation for when a new user signs up for the app.

**Actual changes**
- Implemented autocomplete for the book title search bar
- Added a confetti animation for when a new user signs up for the app
- Made it so that the view moves up when the keyboard appears
- Added tap gestures to help toggle the keyboard

**Test plan**

autocomplete

Confetti

### CODE REVIEW

File: BookWorm/BookWorm/View Controllers/ProfileViewController.m [hunk 1]
```diff
@@ -97,14 +101,35 @@ - (PFFileObject *)getPFFileFromImage: (UIImage * _Nullable)image {
...
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardWillHide:) name:UIKeyboardWillHideNotification object:nil];
+}
+
+- (void)viewWillDisappear:(BOOL)animated {
+    [super viewWillDisappear:animated];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIKeyboardWillShowNotification object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIKeyboardWillHideNotification object:nil];
+}
+
+#pragma mark - keyboard movements
```
Comment (fbxidong) [inline BookWorm/BookWorm/View Controllers/ProfileViewController.m:116]: What a neat feature!

File: BookWorm/BookWorm/View Controllers/SearchViewController.m [hunk 2]
```diff
@@ -107,32 +195,51 @@ -(NSString *)splitSearchWords:(NSString *)wordsToSplit{
...
-    
-    cell.author.text = authors;
-    cell.bookDescription.text = book.subtitle;
+    if (tableView == self.tableView) {
+        BookCell *cell = [tableView dequeueReusableCellWithIdentifier:@"BookCell"];
+        // Configuring the BookCell
+        GoogleBook *book = self.arrayOfBooks[indexPath.row];
+        cell.title.text = book.title;
+        
+        NSString *authors = @"";
```
Comment (fbxidong) [inline BookWorm/BookWorm/View Controllers/SearchViewController.m]: nit: maybe we could wrap the logic of generating authors string into a helper method, which helps to make the cell content logic cleaner

**Review by fbxidong — APPROVED** [review]
Look good to me!

File: BookWorm/BookWorm/View Controllers/SearchViewController.m [hunk 3]
```diff
@@ -27,7 +29,90 @@ - (void)viewDidLoad {
...
+    NSString *googleKey = [dict objectForKey: @"GoogleBooksAPIKey"];
+    NSString *finalURL = [titleString stringByAppendingFormat:@"%@%@", @"&key=", googleKey];
+    NSURL *url = [NSURL URLWithString:finalURL];
+    NSLog(@"Final URL");
+    NSLog(@"%@", finalURL);
+    NSURLRequest *request = [NSURLRequest requestWithURL:url cachePolicy:NSURLRequestReloadIgnoringLocalCacheData timeoutInterval:10.0];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:nil delegateQueue:[NSOperationQueue mainQueue]];
+    NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+           if (error != nil) {
+               NSLog(@"%@", [error localizedDescription]);
```
Comment (Legume1728) [inline BookWorm/BookWorm/View Controllers/SearchViewController.m:81]: You can short-circuit the error, so the rest of this code block isn't as nested.

**Comment by Legume1728** [comment]
Looks good!

**Pull request merged**